packages/mds-agency/handlers/create-event.ts

```diff
@@ -122,7 +122,10 @@ export const createEventHandler = async (
     })()
 
     // TODO switch to cache for speed?
-    const device = await db.readDevice(device_id, provider_id)
+    const device = await IngestServiceClient.getDevice({ device_id, provider_id })
+    if (!isDefined(device)) {
+      throw new NotFoundError(`device_id ${device_id} not found`)
```

Comment:
We have identical code like this introduced in 3 places by this PR. Any practical way to refactor?

Reply:
Good question. I originally thought of throwing inside the repository method, but that might not always be desirable behavior. Sometimes you might not want to throw just because the device wasn't there. Also, I wouldn't want what should be a `NotFoundError` getting rethrown as a `RepositoryError`--I'm a fan of having the exception class give you info about the exception.

Reply:
Ooh, you could try throwing it from the Service Provider, then the error will be able to retain its details 👀